base/broadcast.jl

```diff
@@ -280,7 +280,10 @@ end
 eltypestuple(a) = (Base.@_pure_meta; Tuple{eltype(a)})
 eltypestuple(T::Type) = (Base.@_pure_meta; Tuple{Type{T}})
 eltypestuple(a, b...) = (Base.@_pure_meta; Tuple{eltypestuple(a).types..., eltypestuple(b...).types...})
-_broadcast_eltype(f, A, Bs...) = Base._return_type(f, eltypestuple(A, Bs...))
```

Comment:
`_broadcast_eltype` does not return `broadcast`'s return eltype, only a guess; in general determining `broadcast`'s return eltype requires performing work equivalent to the `broadcast` operation itself. Please find additional discussion in  https://github.com/JuliaLang/julia/pull/19937#issuecomment-271186388. Best!